### PR TITLE
Adding dependancies dockerfile

### PR DESCRIPTION
**Pull request by Shimwell**
## Proposed changes

This adds automated build of a new dependencies image that can be build with each push to the main branch. Once this has been built then it can be used in the circle ci workflow.

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Code refactoring
- [ ] Documentation Update (if none of the other choices apply)
- [ ] New tests

## Checklist

- [ ] Pep8 applied
- [ ] Unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by Shimwell** [comment]
@RemDelaporteMathurin I recall that you asked for something like this a little while back. I know it is not quite ideal as a second Dockerfile was needed. But it is better than having a 2nd repo with the dockerfile.

**Comment by RemDelaporteMathurin** [comment]
Ah nice! I guess it's a rather good workaround while the conditional COPY command isn't available. Is it possible to build this dependencies image on push to dev (or main) and push it to Dockerhub ?

File: .github/workflows/dockerhub-publish-dependencies.yml [hunk 1]
```diff
@@ -0,0 +1,41 @@
+# This yml file will trigger a Github Action on pushes to the main branch.
+# This Action will build and upload a Docker image to Dockerhub
+# https://github.com/marketplace/actions/publish-docker
+
+name: dockerhub-publish-dependencies
+
+on:
+  push:
+    branches: main
```
Comment (Shimwell) [inline .github/workflows/dockerhub-publish-dependencies.yml:9]:
```suggestion
    branches:
    - develop  
    - main
```

**Comment by Shimwell** [comment]
Now build the dependencies image on a push to main or develop

**Pull request merged**